src/ui/components/common/content_actions/comment_count.rs

```diff
@@ -0,0 +1,25 @@
+use crate::{
+  ui::components::common::icon::{Icon, IconType},
+  utils::types::Comments,
+};
+use leptos::*;
+use leptos_router::A;
+
+#[component]
+pub fn CommentCount(id: i32) -> impl IntoView {
+  let comments = expect_context::<Signal<Comments>>();
+  let num_comments_label = Signal::derive(move || format!("{} comments", comments.get().0));
```

Comment:
The comment count pry shouldn't be part of the context right, just a signal input. Also the param should pry be `post_id: PostId`, to be clear.

Reply:
I moved up the comment count to the post listing. This required that I move the wrapping `div` around the content actions up to the post listing, but this is a small inconvenience compared to using the context.